sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgBuilder_Instructions.cs

```diff
@@ -1533,6 +1533,201 @@ public string Foo(Class1 c)
                 // %0 := object.ToString() [ c ]
         }
     }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+
+        [TestMethod]
+        public void AddAssignment_LocalVariable()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        public string fieldString;
+        public string PropertyString { get; set; }
+
+        public void Foo(string parameterString)
+        {
+            var localString = ""foo"";
+                // localString := __id [ const ]
+
+            localString += ""foo"";
+                // %0 := __concat [ localString const ]
+                // localString := __id [ %0 ]
+
+            localString += localString;
+                // %1 := __concat [ localString localString ]
+                // localString := __id [ %1 ]
+
+            localString += parameterString;
+                // %2 := __concat [ localString parameterString ]
+                // localString := __id [ %2 ]
+
+            localString += fieldString;
+                // %4 := __id [ this.fieldString ]
+                // %3 := __concat [ localString %4 ]
+                // localString := __id [ %3 ]
+
+            localString += PropertyString;
+                // %6 := Namespace.Class1.PropertyString.get [ this ]
+                // %5 := __concat [ localString %6 ]
+                // localString := __id [ %5 ]
+
+            Foo(localString += ""abc"");
+                // %7 := __concat [ localString const ]
+                // localString := __id [ %7 ]
+                // %8 := Namespace.Class1.Foo(string) [ this localString ]
+
+            localString += localString += localString += ""123"";
```

Comment:
This test case also applies to fields and properties, doesn't it?

Reply:
Same for the test above - _Foo(localString += ...)_

Reply:
Do you think it's worth adding it for all?

Reply:
We expect the instructions generated for fields and properties to be different, so yes (although probably two += chained together would be enough).

Also, our translation code is context-sensitive so it matters what the syntax nodes before and after are. Most of the bugs we've had so far are because we didn't expect a node of type _X_ at this point.